lib/ipam/ipam.go

```diff
@@ -320,6 +329,12 @@ func (c ipamClient) autoAssign(ctx context.Context, num int, handleID *string, a
 		rem := num - len(ips)
 		retries := ipamEtcdRetries
 		for rem > 0 && retries > 0 {
+			if maxNumBlocks > 0 && numBlocksOwned >= maxNumBlocks {
+				log.WithError(err).Errorf("Unable to allocate a new block, hit blocks per host limit (%v)",
```

Comment:
log.WithError(err) here isn't quite right, right? Since we're not handling an error returned from elsewhere.

Also, would be nice to include numBlocksOwned as well, just in case we somehow get a weird situation where we own say 5 blocks but the limit is 1.

Reply:
Could do:

```
fields := log.Fields{"limit": maxNumBlocks, "owned": numBlocksOwned}
log.WithFields(fields).WithError(ErrBlockLimit).Errorf("unable to allocate block")
```